File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/domain/stateless/StatelessEntityTest.java
package com.coupang.springframework.data.requery.domain.stateless;

import com.coupang.springframework.data.requery.domain.AbstractDomainTest;
import org.junit.Test;

import java.util.UUID;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author Diego on 2018. 6. 12..
 */
public class StatelessEntityTest extends AbstractDomainTest {

    @Test
    public void insert_and_delete_stateless() {

        UUID uuid = UUID.randomUUID();

        Entry entry = new Entry();
        entry.setId(uuid.toString());
        entry.setFlag1(true);
        entry.setFlag2(false);

        requeryTemplate.insert(entry);

        Entry found = requeryTemplate.findById(Entry.class, entry.getId());
        assertThat(found.getId()).isEqualTo(entry.getId());
    }
}

File: src/test/kotlin/com/coupang/springframework/data/requery/domain/stateless/StatelessEntityTest.kt
package com.coupang.springframework.data.requery.domain.stateless

import com.coupang.springframework.data.requery.domain.AbstractDomainTest
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import java.util.*

/**
 * StatelessEntityTest
 *
 * @author debop@coupang.com
 * @since 18. 5. 31
 */
class StatelessEntityTest: AbstractDomainTest() {

    @Test
    fun `insert and delete stateless`() {

        val uuid = UUID.randomUUID()

        val entity = StatelessEntity().also {
            it.id = uuid.toString()
            it.flag1 = true
            it.flag2 = false
        }

        requeryTmpl.insert(entity)

        val found = requeryTmpl.findById(StatelessEntity::class.java, entity.id)!!
        assertThat(found.id).isEqualTo(entity.id)
    }
}